cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -498,16 +512,22 @@ public void getArtifactPlugin(HttpRequest request, HttpResponder responder,
                                 @PathParam("artifact-version") String artifactVersion,
                                 @PathParam("plugin-type") String pluginType,
                                 @PathParam("plugin-name") String pluginName,
-                                @QueryParam("scope") @DefaultValue("user") String scope)
-    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException {
+                                @QueryParam("scope") @DefaultValue("user") String scope,
+                                @QueryParam("versionRange") String versionRange,
```

Comment:
for example, somebody is using plugin 'DB' of type 'batchsource', which extends cdap-data-pipeline-4.2.0. If there is a DB plugin from db-plugins-1.7.0 and one from my-custom-plugins-1.7.0, you need to be able to specify which one you're interested in.